File: frontend/src/microservices/advertising/objects/UploadAdvertisementImageResponse.ts
// 本文件定义 advertising 模块的 `UploadAdvertisementImageResponse`，用于图片上传响应并提供 JSON 编解码。

export type UploadAdvertisementImageResponse = {
  assetId: string
  publicUrl: string
  originalFileName: string
  mimeType: string
  fileSize: number
}

export const uploadAdvertisementImageResponseFromJson = (json: string): UploadAdvertisementImageResponse =>
  JSON.parse(json) as UploadAdvertisementImageResponse

export const uploadAdvertisementImageResponseToJson = (value: UploadAdvertisementImageResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/UploadAdvertisementImageResponse.scala
// 本文件定义 advertising 模块的 UploadAdvertisementImageResponse，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadAdvertisementImageResponse(
    assetId: String,
    publicUrl: String,
    originalFileName: String,
    mimeType: String,
    fileSize: Long
)

object UploadAdvertisementImageResponse:
  given sourceEncoder: Encoder[UploadAdvertisementImageResponse] = deriveEncoder
  given sourceDecoder: Decoder[UploadAdvertisementImageResponse] = deriveDecoder

